src/util/fileinfo.h

```diff
@@ -255,6 +257,13 @@ class FileInfo final {
         return m_fileInfo.size();
     }
 
+    // This can be used to assert that the object is thread-safe copyable like QFileInfo
+    static constexpr bool isQFileInfo {
+        // Additional member variables will violate this assumption
+        return (sizeof(QFileInfo) == sizeof(mixxx::FileInfo) &&
+                std::is_same_v<decltype(m_fileInfo), QFileInfo>)
+    }
```

Comment:
```suggestion
    // This only works because Qt's implicit sharing allows copies to be threadsafe and
    // this class only consists of a single QFileInfo member.
    // Adding additional member variables will likely violate this assumption.
    static constexpr bool canBeCopiedThreadSafe = sizeof(QFileInfo) == sizeof(mixxx::FileInfo) && std::is_same_v<decltype(m_fileInfo), QFileInfo>
    }
```